File: sic-api/Entities/Db/DbMailTemplate.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace sic_api.Entities.Db;

[Index(nameof(TemplateCode), IsUnique = true)]
[Table("db_mail_template")]
public class DbMailTemplate : BaseEntity
{
    [Required]
    [MaxLength(50)]
    [Column("template_code")]
    public string TemplateCode { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("template_name")]
    public string TemplateName { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("subject_en")]
    public string SubjectEn { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("subject_local")]
    public string SubjectLocal { get; set; } = default!;

    [Required]
    [Column("content_en")]
    public string ContentEn { get; set; } = default!;

    [Required]
    [Column("content_local")]
    public string ContentLocal { get; set; } = default!;

    [Column("is_html")]
    public bool IsHtml { get; set; } = false;

    [Column("is_active")]
    public bool IsActive { get; set; } = false;

    [MaxLength(3000)]
    [Column("variables")]
    public string? Variables { get; set; }

    public ICollection<DbMailQueue>? MailQueues { get; set; }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/db/DbMailTemplate.java
package com.softinter.sicapi.entity.db;

import com.softinter.sicapi.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "db_mail_template",
       indexes = {
           @Index(name = "idx_template_code", columnList = "template_code", unique = true)
       })
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class DbMailTemplate extends BaseEntity {

    @Column(name = "template_code", nullable = false, length = 50)
    private String templateCode;

    @Column(name = "template_name", nullable = false, length = 255)
    private String templateName;

    @Column(name = "subject_en", nullable = false, length = 255)
    private String subjectEn;

    @Column(name = "subject_local", nullable = false, length = 255)
    private String subjectLocal;

    @Column(name = "content_en", nullable = false, columnDefinition = "TEXT")
    private String contentEn;

    @Column(name = "content_local", nullable = false, columnDefinition = "TEXT")
    private String contentLocal;

    @Column(name = "is_html")
    private Boolean isHtml = false;

    @Column(name = "is_active")
    private Boolean isActive = false;

    @Column(name = "variables", length = 3000)
    private String variables;

    @OneToMany(mappedBy = "mailTemplate") // assuming DbMailQueue has a field 'mailTemplate'
    private List<DbMailQueue> mailQueues = new ArrayList<>();
}